RIOT-OS/RIOT
# cpu/esp32: split the doc into ESP32x common and ESP32 specific parts

### Contribution description

This PR is a split-off from #17842, providing the separation of of the documentation into a part common to all ESP32x SoCs and a part specific to ESP32 to support different ESP32x SoC variants.

### Testing procedure

1. Green CI
2. `make doc` and check that there is a new sub-menu `RIOT OS -> Modules -> CPU -> ESP32x SoC Series -> ESP32 family` which contains specific information about ESP32.

### Issues/PRs references

Split-off from #17842
